### PR TITLE
Forms

### PR DESCRIPTION
**Pull request by zhorabay**
- [ ] In this branch, I:
- Added a method called current_user that will return the first user from the database. to make current user data available to all controllers
- Created forms that
-> creates a Post on behalf of the current_user 
-> creates a Comment on behalf of the current_user
-> adds Likes to Posts

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -7,4 +10,33 @@ def index
...
+    params.require(:post).permit(:title, :text)
+  end
+
+  def find_post
+    @post = Post.find(params[:id])
+  end
+
+  def current_user
+    @current_user = User.find(session[:user_id]) if session[:user_id]
+  end
```
Comment (mikethreels) [inline app/controllers/posts_controller.rb]: - [x] please use the application_controller for this method so it is available in the controllers.

File: app/controllers/posts_controller.rb [hunk 2]
```diff
@@ -7,4 +10,33 @@ def index
...
+      redirect_to user_post_path(@user, @post), notice: 'Post created successfully.'
+    else
+      render :new, status: :unprocessable_entity
+    end
+  end
+
+  private
+
+  def post_params
+    params.require(:post).permit(:title, :text)
```
Comment (mikethreels) [inline app/controllers/posts_controller.rb]: - [ ] please make sure the post is created on behalf of the current_user which is not the case at the moment.

File: app/controllers/comments_controller.rb [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    @post = Post.find(params[:post_id])
+    @comment = @post.comments.build(comment_params.merge(post: @post))
+
+    if @comment.save
+      @post.update(comments_counter: @post.comments.count)
+      redirect_to user_post_path(@post.author, @post)
+    else
+      redirect_to user_post_path(@post.author, @post), alert: 'Unable to add the comment.'
+    end
+  end
```
Comment (mikethreels) [inline app/controllers/comments_controller.rb]: - [ ] please make sure the comment is created on behalf of the current logged in user.

File: app/controllers/likes_controller.rb [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+
+  def create
+    @post = Post.find(params[:post_id])
+    @like = @post.likes.new(user: @user)
+
+    if @like.save
+      @post.update(likes_counter: @post.likes.count)
+      redirect_to user_post_path(@post.author, @post), notice: 'You liked the post!'
+    else
+      redirect_to user_post_path(@post.author, @post), alert: 'Unable to like the post.'
```
Comment (mikethreels) [inline app/controllers/likes_controller.rb]: - [ ] please make sure the like is created on behalf of the currently logged in user.

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi Assem, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- Good job creating the "create" methods.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/posts_controller.rb (hunk 1 above)
Comment (zhorabay) [reply app/controllers/posts_controller.rb:41]: Oops, now I've got it. I was wondering why my code was not working properly. I looked at others' code and even copied some, thinking I was writing it wrong all the time. Thank you for your review :)

Comment (Whoistolu) [reply app/controllers/posts_controller.rb]: - Well resolved

File: app/controllers/posts_controller.rb (hunk 2 above)
Comment (Whoistolu) [reply app/controllers/posts_controller.rb]: - Well resolved

File: app/controllers/comments_controller.rb (hunk 3 above)
Comment (Whoistolu) [reply app/controllers/comments_controller.rb]: - Well resolved

File: app/controllers/likes_controller.rb (hunk 4 above)
Comment (Whoistolu) [reply app/controllers/likes_controller.rb]: - Well resolved

**Review by Whoistolu — APPROVED** [review]
### Hi @zhorabay 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

# STATUS: APPROVED 🟢
**Congratulations! 🎉** 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @whoistolu in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**